internal/sink/sink.go

```diff
@@ -123,6 +130,8 @@ func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
 				}
 			}
 
+			whs.lock.Lock() //QUESTION: would this go here or inside the goroutine? if here do we need to change signature of this function?
+			defer whs.lock.Unlock()
```

Comment:
We'll incorporate a lock soon, it'll be used during when the sink is either updated, started or shutdown